src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,144 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.jdbc.custromexception.CustomJdbcException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String ID = "id";
+    private static final String NAME = "name";
+    private static final String COUNTRY = "country";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new CustomJdbcException("Wasn't added a new instance with parameters:"
+                    + System.lineSeparator()
+                    + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = null;
+        String getManufacturerRequest =
+                "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getInstanceManufacturersStatement = connection.prepareStatement(
+                        getManufacturerRequest)) {
+            getInstanceManufacturersStatement.setLong(1, id);
+            ResultSet resultSet = getInstanceManufacturersStatement.executeQuery();
+            if (resultSet.next()) {
+                manufacturer = createManufacturerFromDataDB(resultSet);
+            } else {
+                throw new CustomJdbcException(("The manufacturer with id = "
+                        + id + " doesn't exists!"));
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new CustomJdbcException("Can't get an instance from DB with id = " + id, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't get an instance from DB with id = " + id, e);
```
![image](https://user-images.githubusercontent.com/58442482/135468529-6dcb64a3-035d-4a4a-8922-fa2fa16540ac.png)